[readme] update M1 to “Apple Silicon” because of M2 and M3; fix typos (#3242)
nvm-sh/nvm: README.md

If the above doesn't fix the problem, you may try the following:

  - If you use bash, it may be that your `.bash_profile` (or `~/.profile`) does not source your `~/.bashrc` properly. You could fix this by adding `source ~/<your_profile_file>` to it or following the next step below.

  - Try adding [the snippet from the install section](#profile_snippet), that finds the correct nvm directory and loads nvm, to your usual profile (`~/.bash_profile`, `~/.zshrc`, `~/.profile`, or `~/.bashrc`).

  - For more information about this issue and possible workarounds, please [refer here](https://github.com/nvm-sh/nvm/issues/576)

> **Note** For Macs with the Apple Silicon chip, node started offering **arm64** arch Darwin packages since v16.0.0 and experimental **arm64** support when compiling from source since v14.17.0. If you are facing issues installing Node using `nvm`, you may want to update to one of those versions or later.

Reviewer: ```suggestion
**Note** For Macs with the Apple Silicon chip, node started offering **arm64** arch Darwin packages since v16.0.0 and experimental **arm64** support when compiling from source since v14.17.0. If you are facing issues installing node using `nvm`, you may want to update to one of those versions or later.
```

Author: Are you sure?? This is some kind of typo and users that use extensions on their browser will noticed that

Reviewer: It’s not a typo, the branding requires lowercase node, or “Node.js”. Nothing else is correct.

### Important Notes

> If you're running a system without prepackaged binary available, which means you're going to install Node Js or io.js from its source code, you need to make sure your system has a C++ compiler. For OS X, Xcode will work, for Debian/Ubuntu based GNU/Linux, the `build-essential` and `libssl-dev` packages work.

Reviewer: ```suggestion
If you're running a system without prepackaged binary available, which means you're going to install node or io.js from its source code, you need to make sure your system has a C++ compiler. For OS X, Xcode will work, for Debian/Ubuntu based GNU/Linux, the `build-essential` and `libssl-dev` packages work.
```

Author: Same here

**Note:** We still have some problems with FreeBSD, because there is no official pre-built binary for FreeBSD, and building from source may need [patches](https://www.freshports.org/www/node/files/patch-deps_v8_src_base_platform_platform-posix.cc); see the issue ticket:

>   - [[#900] [Bug] Node Js on FreeBSD may need to be patched](https://github.com/nvm-sh/nvm/issues/900)

Reviewer: ```suggestion
  - [[#900] [Bug] node on FreeBSD may need to be patched](https://github.com/nvm-sh/nvm/issues/900)
```

Homebrew causes insecure directories like `/usr/local/share/zsh/site-functions` and `/usr/local/share/zsh`. This is **not** an `nvm` problem - it is a homebrew problem. Refer [here](https://github.com/zsh-users/zsh-completions/issues/680) for some solutions related to the issue.

**Macs with Apple Silicon chips**

Experimental support for the Apple Silicon chip architecture was added in node.js v15.3 and full support was added in v16.0.
Because of this, if you try to install older versions of node as usual, you will probably experience either compilation errors when installing node or out-of-memory errors while running your code.

> So, if you want to run a version before v16.0 on an Apple Silicon Mac, it may be best to compile Node Js targeting the `x86_64` Intel architecture so that Rosetta 2 can translate the `x86_64` processor instructions to ARM-based Apple Silicon instructions.

Reviewer: ```suggestion
So, if you want to run a version prior to v16.0 on an Apple Silicon Mac, it may be best to compile node targeting the `x86_64` Intel architecture so that Rosetta 2 can translate the `x86_64` processor instructions to ARM-based Apple Silicon instructions.
```

  $ arch -x86_64 zsh
  ```

>   Note: This same thing can also be accomplished by finding the Terminal or iTerm App in Finder, right click on it, selecting "Get Info", and then checking the box labeled "Open using Rosetta".

Reviewer: ```suggestion
  Note: This same thing can also be accomplished by finding the Terminal or iTerm App in Finder, right clicking, selecting "Get Info", and then checking the box labeled "Open using Rosetta".
```
it has to be ing to match the other two.

  $ source "${NVM_DIR}/nvm.sh"
  ```

> - Install whatever older version of Node you are interested in. Let's use 12.22.1 as an example.

Reviewer: ```suggestion
- Install whatever older version of node you are interested in. Let's use 12.22.1 as an example.
```
